Use in-memory cache for autoscale HA cluster

programmerq opened the pull request:
Fixes #4894

programmerq committed:
Use in-memory cache for autoscale HA cluster

Fixes #4894

programmerq committed:
Merge branch 'master' into issue_4894

russjones commented:
@webvictim Can you take a look?

webvictim commented on assets/aws/files/bin/teleport-generate-config:
```diff
@@ -255,6 +255,11 @@ teleport:
+  cache:
+    type: in-memory
+  connection_limits:
+    max_connections: 6500
```
This should be 10x as per https://community.goteleport.com/t/memory-cache-for-proxy/882

```suggestion
    max_connections: 65000
```

webvictim reviewed (commented):
This will also affect the `starter-cluster` as well as `ha-autoscale-cluster` - not sure if that's the intention?

I'd appreciate if you could also add some unit tests (similar to https://github.com/gravitational/teleport/blob/master/assets/aws/files/tests/ha-auth.bats#L26-L31) which make sure that:
- `cache.type` is set to `in-memory`
- `connection_limits.max_connections` is set to 65000
- `connection_limits.max_users` is set to 10000

This will help prevent a potential regression in future if changes are made to the config generation script.

programmerq commented:
I'll add in the unit tests-- I didn't realize the BATS tests were there! Fantastic!

I'll do some thinking on how to do this only for the ha cluster but not the starter cluster.

webvictim commented:
> I'll do some thinking on how to do this only for the ha cluster but not the starter cluster.

It’s not super critical honestly, I think it’s fine to just let it affect both. We’d probably need to add an additional variable to each Terraform and I don’t think it’s really that useful. The in-memory cache isn’t ever really a bad thing as far as I know.

programmerq committed:
Merge branch 'master' into issue_4894

programmerq committed:
bump max_connections and add BATS test

programmerq commented:
* I added a BATS test for the ha cluster
* I updated it to 6500
* I noticed that the script already distinguishes between the proxy only role (used by HA) and the auth,proxy,node role (used by starter). This means that the change only applies to the ha mode proxy component like originally intended.

webvictim committed:
Merge branch 'master' into issue_4894

russjones reviewed (approved):
Bot.

webvictim committed:
Merge branch 'master' into issue_4894

webvictim committed:
Merge branch 'master' into issue_4894

russjones committed:
Merge branch 'master' into issue_4894

russjones committed:
Merge branch 'master' into issue_4894

russjones committed:
Merge branch 'master' into issue_4894

programmerq committed:
Merge branch 'master' into issue_4894

programmerq committed:
Merge branch 'master' into issue_4894

programmerq committed:
Merge branch 'master' into issue_4894

russjones committed:
Merge branch 'master' into issue_4894

programmerq committed:
Merge branch 'master' into issue_4894

russjones commented:
Will merge in https://github.com/gravitational/teleport/pull/9323.
